tests/ObjectPoolTests.cpp

```diff
@@ -0,0 +1,146 @@
+#include "doctestUtils.h"
+
+#include <cesium/omniverse/ObjectPool.h>
+#include <doctest/doctest.h>
+#include <sys/types.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <queue>
+
+#define MAX_TESTED_POOL_SIZE 1024 // the max size pool to randomly generate
```

Comment:
Style: prefer `constexpr` over defines for compile time constants.

`constexpr uint64_t MAX_TESTED_POOL_SIZE = 1024;`